xds/internal/xdsclient/xds.go

```diff
@@ -812,18 +812,39 @@ func securityConfigFromCluster(cluster *v3clusterpb.Cluster) (*SecurityConfig, e
 		return nil, errors.New("UpstreamTlsContext in CDS response does not contain a CommonTlsContext")
 	}
 
-	sc, err := securityConfigFromCommonTLSContext(upstreamCtx.GetCommonTlsContext())
-	if err != nil {
-		return nil, err
+	return securityConfigFromCommonTLSContext(upstreamCtx.GetCommonTlsContext(), false)
+}
+
+// common is expected to be not nil.
+func securityConfigFromCommonTLSContext(common *v3tlspb.CommonTlsContext, server bool) (*SecurityConfig, error) {
+	sc := securityConfigFromCommonTLSContextUsingNewFields(common)
+	var err error
+	if sc == nil || sc.Equal(&SecurityConfig{}) {
```

Comment:
Why does the function return `nil` or an empty `SecurityConfig`? Can it just return `nil`?

Reply:
See other comment.